File: Dot examples/ASP Book Examples/Book Examples/Chapter28/VB/App_Code/DesignSection.vb
Imports System
Imports System.Configuration
Imports System.Drawing

Namespace AspNetUnleashed
    Public Class DesignSection
        Inherits ConfigurationSection

        <ConfigurationProperty("backcolor", DefaultValue:="lightblue", IsRequired:=True)> _
        Public Property BackColor() As Color
            Get
                Return CType(Me("backcolor"), Color)
            End Get
            Set(ByVal Value As Color)
                Me("backcolor") = Value
            End Set
        End Property

        <ConfigurationProperty("styleSheetUrl", DefaultValue:="~/styles/style.css", IsRequired:=True)> _
        <RegexStringValidator(".css$")> _
        Public Property StyleSheetUrl() As String
            Get
                Return CType(Me("styleSheetUrl"), String)
            End Get
            Set(ByVal Value As String)
                Me("styleSheetUrl") = Value
            End Set
        End Property

        Public Sub New(ByVal backcolor As Color, ByVal styleSheetUrl As String)
            Me.BackColor = backcolor
            Me.StyleSheetUrl = styleSheetUrl
        End Sub

        Public Sub New()
        End Sub
    End Class
End Namespace

File: Dot examples/ASP Book Examples/Book Examples/Chapter28/CS/App_Code/DesignSection.cs
using System;
using System.Configuration;
using System.Drawing;

namespace AspNetUnleashed
{
    public class DesignSection : ConfigurationSection
    {
        [ConfigurationProperty("backcolor", DefaultValue = "lightblue", IsRequired = true)]
        public Color BackColor
        {
            get { return (Color)this["backcolor"]; }
            set { this["backcolor"] = value; }
        }

        [ConfigurationProperty("styleSheetUrl", DefaultValue = "~/styles/style.css", IsRequired = true)]
        [RegexStringValidator(".css$")]
        public string StyleSheetUrl
        {
            get { return (string)this["styleSheetUrl"]; }
            set { this["styleSheetUrl"] = value; }
        }

        public DesignSection(Color backcolor, string styleSheetUrl)
        {
            this.BackColor = backcolor;
            this.StyleSheetUrl = styleSheetUrl;
        }

        public DesignSection()
        {
        }
    }
}